fix(api): dispatcher.integration test breaks on 7-day window boundary (hardcoded 2026-04-20 date)

## Goal

Fix the time-windowed test in `packages/api/tests/dispatcher.integration.test.ts:1066` ("seeded diagnoses within 7 days are aggregated...") so it doesn't break on the SQL `now() - interval '7 days'` boundary.

## Symptom

As of 2026-04-27 ~12:00 UTC, the test seeds `completedAt = '2026-04-20T12:00:00Z'` and expects rows with that day to be returned by `weeklyDiagnoserReport`. The resolver (`packages/api/src/graphql/dispatcher/resolvers.ts:265`) filters `WHERE completed_at >= now() - interval '7 days'`. By 2026-04-27T14:00Z the seeded timestamp has slipped just outside the 7-day window, the resolver returns 0 matching rows, and:

```
AssertionError: expected undefined not to be undefined
 ❯ tests/dispatcher.integration.test.ts:1125:29
```

Verified on PR #3572 ([run 24999056386](https://github.com/judgemind/judgemind/actions/runs/24999056386), [run 24999098916](https://github.com/judgemind/judgemind/actions/runs/24999098916)) — both api-tests jobs fail at line 1125 with this exact error. Main-branch CI hasn't tripped because no main run has executed `api-tests` (non-skipped) since the date crossed.

## Root cause

Hardcoded `dayTs = '2026-04-20T12:00:00Z'` at `packages/api/tests/dispatcher.integration.test.ts:1075`. Same hardcoding probably exists at sibling tests in the same describe block.

## Fix shape

Replace the literal with a relative timestamp:

```ts
const dayTs = new Date(Date.now() - 3 * 24 * 60 * 60 * 1000).toISOString();
// 3 days ago — well inside the 7-day window, well outside the >=8-days exclusion test
```

Audit the rest of `dispatcher.integration.test.ts` for similar hardcoded `2026-04-` and `2026-04-20T12:00:00Z` literals. The test pattern uses both "within 7 days" and ">8 days ago" assertions; both must use relative timestamps.

## Acceptance criteria

- [ ] No hardcoded calendar dates in `packages/api/tests/dispatcher.integration.test.ts` for `weeklyDiagnoserReport` or any other 7-day-window test. Verify: `grep -n "2026-04-" packages/api/tests/dispatcher.integration.test.ts` returns 0 matches.
- [ ] `cd packages/api && npm test -- dispatcher.integration` passes locally.
- [ ] CI green on the resulting PR.

## Priority

**p1** — currently blocks every PR that triggers `api-tests` (any change to `packages/api`, `packages/web`, or `.github/workflows/ci.yml`). Will cascade through the dispatcher daemon's queue today as more agents try to ship.

## Notes

- Discovered via PR #3572 (docs spec/SKILL.md alignment). That PR doesn't touch `packages/api` but does touch `.github/workflows/ci.yml`, which forces api-tests to run.
- I'll merge #3572 with admin override or use a follow-up rebase once this lands; the api-tests failure is unrelated to its scope.

type/dx · area/devops · priority/p1 · agent/ready

--- a/packages/api/tests/dispatcher.integration.test.ts
+++ b/packages/api/tests/dispatcher.integration.test.ts
@@ -702,6 +702,7 @@ describe('queueReady — SQL predicate functions contract (#3001)', () => {
   const ISSUE_B = 1002;
   const ISSUE_C = 1003;
   const insertedAgentIds3001: string[] = [];
+  const fixtureCreatedAt = new Date(Date.now() - 30 * 24 * 60 * 60 * 1000).toISOString();
 
   beforeAll(async () => {
     // Insert a run for the snapshot.
@@ -716,9 +717,9 @@ describe('queueReady — SQL predicate functions contract (#3001)', () => {
 
     // Insert queue snapshot with issues A, B, C.
     const issuesJson = JSON.stringify([
-      { number: ISSUE_A, title: 'Issue A (never attempted)', labels: ['priority/p2', 'agent/ready'], createdAt: '2026-04-01T00:00:00Z' },
-      { number: ISSUE_B, title: 'Issue B (running agent)', labels: ['priority/p2', 'agent/ready'], createdAt: '2026-04-01T00:00:00Z' },
-      { number: ISSUE_C, title: 'Issue C (failed, in cooldown)', labels: ['priority/p2', 'agent/ready'], createdAt: '2026-04-01T00:00:00Z' },
+      { number: ISSUE_A, title: 'Issue A (never attempted)', labels: ['priority/p2', 'agent/ready'], createdAt: fixtureCreatedAt },
+      { number: ISSUE_B, title: 'Issue B (running agent)', labels: ['priority/p2', 'agent/ready'], createdAt: fixtureCreatedAt },
+      { number: ISSUE_C, title: 'Issue C (failed, in cooldown)', labels: ['priority/p2', 'agent/ready'], createdAt: fixtureCreatedAt },
     ]);
     await pool.query(
       `INSERT INTO dispatcher.queue_snapshots (observed_at, queue_depth, issue_numbers, issues_json, run_id)
@@ -834,6 +835,7 @@ describe('dispatcherQueueFull — admin (#3159)', () => {
   const READY_BASE = 31590;
   const BLOCKED_BASE = 31700;
   const COMPLETED_AGENT_BASE = `dqf-3159-${MARKER}-`;
+  const fixtureCreatedAt = new Date(Date.now() - 30 * 24 * 60 * 60 * 1000).toISOString();
 
   beforeAll(async () => {
     queueRunId = await insertRun();
@@ -844,7 +846,7 @@ describe('dispatcherQueueFull — admin (#3159)', () => {
       number: READY_BASE + i,
       title: `ready test ${i}`,
       labels: ['priority/p2', 'agent/ready'],
-      createdAt: '2026-04-01T00:00:00Z',
+      createdAt: fixtureCreatedAt,
     }));
     await pool.query(
       `INSERT INTO dispatcher.queue_snapshots
@@ -863,7 +865,7 @@ describe('dispatcherQueueFull — admin (#3159)', () => {
       number: BLOCKED_BASE + i,
       title: `blocked test ${i}`,
       labels: ['priority/p2', 'status/blocked'],
-      createdAt: '2026-04-01T00:00:00Z',
+      createdAt: fixtureCreatedAt,
       body: '',
     }));
     await pool.query(
@@ -1072,7 +1074,7 @@ describe('weeklyDiagnoserReport — admin', () => {
       detectedBy: 'scheduler',
     });
 
-    const dayTs = '2026-04-20T12:00:00Z'; // within 7 days of "now" in tests
+    const dayTs = new Date(Date.now() - 3 * 24 * 60 * 60 * 1000).toISOString(); // within 7 days of now
     // Two 'retry' / 'succeeded' rows on the same day → count 2
     await insertDiagnosis({
       agentId,

